Apply flashcart core

d3m3vilurr opened the pull request:
BTW, still not work.

d3m3vilurr committed:
Remove device related codes

- flashcart_core is better
- btw we need implement part of 3ds related features

d3m3vilurr committed:
Stub implement protocol_ntr

d3m3vilurr committed:
Change reset commands

kitling commented:
This is looking pretty good, however, there a few tweaks to flashcart_core that will need to be made before I merge something like this in.

d3m3vilurr commented:
yes. flashcart_core have 4 outside functions,
`ShowProgress`, `ioDelay`, `Cart_NTRInit` and `NTR_SendCommand`

at least, I think `ShowProgress` should be callback. :)

d3m3vilurr committed:
Update flashcart_core

now, flashcart_core remove & not implement platform specific methods
except ioDelay.
so, remove all gamecart codes, then add platform.cpp for this.

kitling commented:
LGTM! ioDelay is the only thing remaining that should be removed, but your workaround is okay for now. Merging.
